lib/bolt/inventory.rb

```diff
@@ -97,8 +96,12 @@ def update_target(target)
         inv_conf = {}
       end
 
-      conf = Bolt::Util.deep_merge(@config.transport_conf, inv_conf)
-      target.update_conf(conf)
+      # Use Config object to ensure config section is treated consistently with config file
+      conf = @config.deep_clone
+      conf.update_from_inventory(inv_conf)
```

Comment:
I haven't made this backward compatible. I'd be tempted to have Inventory error if it finds a `transports` property in `config` and require people to update their inventory files.

Reply:
ok